include/swift/AST/DiagnosticsCommon.def

```diff
@@ -174,6 +174,11 @@ ERROR(cannot_load_swiftoverlay_file, none,
 ERROR(cannot_list_swiftcrossimport_dir, none,
       "cannot list cross-import overlays for %0: %1 (declared in '%2')",
       (Identifier, StringRef, StringRef))
+WARNING(cross_imported_by_both_modules, none,
+        "modules %0 and %1 both declare module %2 as a cross-import overlay, "
+        "which may cause paradoxical behavior when looking up names in them; "
+        "please report this bug to the maintainers of these modules",
+        (Identifier, Identifier, Identifier))
```

Comment:
I like this diagnostic. It is wordier than our typical diagnostics but it explains things clearly.